git: Fix file history view shows duplicate entries when clicking on `Load More`

relaxcn opened the pull request:
This is because the git command parameter `--skip` doesn't work correctly if it's used with `--follow`.

So I use a cursor-based pagination approach (using commit hash + `^` notation) to reliably fetch the next page of file history.

Closes #44591

Release Notes:

- Fixed file history view shows duplicate entries when clicking on `Load More`.

relaxcn committed:
git: Fix file history view shows duplicate entries when clicking on `Load More`

This is because the git command parameter `--skip` doesn't work
correctly if it's used with `--follow`

cameron1024 reviewed (commented):
Thanks for the contribution 🙏 Small tweak needed but otherwise looking good

relaxcn commented on crates/proto/proto/git.proto:
```diff
@@ -310,7 +310,7 @@ message GitFileHistory {
     uint64 repository_id = 3;
     string path = 4;
-    uint64 skip = 5;
+    optional string last_commit_hash = 5;
```
Thank you.
I will add a new field instead of removing the old one.

orengriffin committed:
Add test for file history pagination with rename

Tests that pagination works correctly when using --follow to track
file history across renames, specifically verifying no duplicate
commits appear across pages.

relaxcn commented:
@orengriffin you need to sign Contributor License Agreement firstly.

Anthony-Eid commented:
closing this because we migrated the file history view to use the git graph under the hood and the git graph fetches the full history at once. Aka, this bug is no longer an issue 😃

Thank you for contributing this fix though!
